lib/bike_brigade_web/components/sms_message_list_component.ex

```diff
@@ -0,0 +1,139 @@
+defmodule BikeBrigadeWeb.Components.SMSMessageListComponent do
+  use Phoenix.Component
+  use BikeBrigadeWeb, :html
+  alias BikeBrigade.Messaging.SmsMessage
+
+
+  def preview_message(%SmsMessage{body: body}) when not is_nil(body) do
+    if String.length(body) > 30 do
+      String.slice(body, 0, 30) <> "..."
+    else
+      body
+    end
+  end
+
+  def preview_message(%SmsMessage{media: media}) when length(media) > 0 do
+    "Attachment: #{length(media)} media"
+  end
+
+  def preview_message(_), do: "unknown message"
+
+
+  def handle_build_message_link({"/messages", rider_id}) do
+    ~p"/messages/#{rider_id}"
+  end
+
+  def handle_build_message_link({"/campaign", campaign_id, rider_id}) do
+    ~p"/campaigns/#{campaign_id}/messaging/riders/#{rider_id}"
+  end
+
+
+  # TODO: can :any be a tuple?
+  # TODO: fix indent.
+  attr :sms_list, :any, required: true
+
+  @doc """
+  A helper component to determine where clicking on a rider in an sms list takes us
+  we render the sms list in different capacities; this allows deciding to which sms
+  instance clicking a rider would take us to.
+  """
+  def message_link(assigns) do
+    route = case assigns.sms_list do
+      {"/messages"} -> ~p"/messages/#{assigns.rider_id}"
+      {"/campaigns", campaign_id} -> ~p"/campaigns/#{campaign_id}/messaging/riders/#{assigns.rider_id}"
+    end
+
+    assigns =  assign(assigns, :route, route)
+
+    ~H"""
+      <.link
+          patch={@route}
+          class=" px-2 py-1 border-b hover:bg-gray-100 block transition duration-150 ease-in-out  focus:outline-none"
+        >
+        <%= render_slot(@inner_block) %>
+      </.link>
+    """
+  end
+
+  @doc"""
+  Renders a list of sms messages
+
+  ## Examples
+      <.sms_message_list
```

Comment:
Example is missing the `sms_list` (or above `rider_link_fn`)

Reply:
Fixed